src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -873,15 +873,18 @@ def matches_service(self, service_name: Optional[str]) -> bool:
     def service_id(self) -> str:
         assert self.daemon_id is not None
         assert self.daemon_type is not None
-        if self.daemon_type == 'osd' and self.osdspec_affinity:
-            return self.osdspec_affinity
 
         if self._service_name:
             if '.' in self._service_name:
                 return self._service_name.split('.', 1)[1]
             else:
                 return ''
 
+        if self.daemon_type == 'osd':
+            if self.osdspec_affinity and self.osdspec_affinity != 'None':
```

Comment:
(nit) seems like a str `'None'` could be problematic elsewhere ...

should `get_osdspec_affinity` return a NoneType?
https://github.com/ceph/ceph/blob/648f9e11a627508296f8f676bd4fde7f78ba27b8/src/pybind/mgr/cephadm/services/osd.py#L298-L299